Sources/Target/Darwin/Thread.cpp

```diff
@@ -33,7 +33,50 @@ namespace Darwin {
 using Host::Darwin::LibProc;
 
 Thread::Thread(ds2::Target::Process *process, ThreadId tid)
-    : super(process, tid), _lastSyscallNumber(-1) {}
+    : super(process, tid), _lastSyscallNumber(-1) {
+  //
+  // Initially the thread is stopped.
+  //
+}
+
+ErrorCode Thread::suspend() {
+  ErrorCode error = kSuccess;
```

Comment:
I think you can simplify this function a lot by doing something like `if (_state == kRunning) return proces()->...suspend(...);` and then
```
if (_state == kTerminated) {
    return kErrorProcessNotFound;
} 

return kSuccess;
```